app/controllers/collections_controller.rb

```diff
@@ -156,12 +156,25 @@ def prepare_translated_collections
     translated_collection_native_xml = cmr_client.translate_collection(JSON.pretty_generate(@collection), "application/#{Rails.configuration.umm_c_version}; charset=utf-8", content_type,  skip_validation=true)
     return { error: 'Failed to translate collection from UMM back to native format' } unless translated_collection_native_xml.success?
 
+    # this checks the 'hide_items' url parameter that is can be manually added. Its primary use is for developers
```

Comment:
If you're going to provide an error for an invalid parameter, I think this should happen before you contact CMR.  Additionally, I don't think this belongs in this function.  

You can do something like 
```
compared collections = if hide_items.is_a?(Hash)
  hide_items
else
  prepare_translated_collections
end
```
compared_collections isn't accurate is it?  They haven't been compared yet?